tensorcircuit/translation.py

```diff
@@ -350,6 +348,16 @@ def qiskit2tc(
             # logger.warning(
             #     "qiskit to tc translation currently doesn't support reset instruction, just skipping"
             # )
+        elif not hasattr(gate_info[0], "__array__"):
+            # an instruction containing a lot of gates.
+            # the condition is based on
+            # https://github.com/Qiskit/qiskit-terra/blob/2f5944d60140ceb6e30d5b891e8ffec735247ce9/qiskit/circuit/gate.py#L43
+            qiskit_circuit = gate_info[0].definition
+            if qiskit_circuit is None:
```

Comment:
I prefer to directly check gate_name == barrier in outer level? which makes the logic more clear. since measure and reset are also in outter level anyway. and we can keep here -- replace the continue to a warning raised, claiming the instruction name is not handled in the translation

Reply:
No problem!